br/pkg/aws/ebs.go

```diff
@@ -49,66 +48,92 @@ func NewEC2Session(concurrency uint) (*EC2Session, error) {
 }
 
 // CreateSnapshots is the mainly steps to control the data volume snapshots.
-// It will do the following works.
-// 1. determine the order of volume snapshot.
-// 2. send snapshot requests to aws.
 func (e *EC2Session) CreateSnapshots(backupInfo *config.EBSBasedBRMeta) (map[string]string, VolumeAZs, error) {
 	snapIDMap := make(map[string]string)
-	volumeIDs := []*string{}
+	var volumeIDs []*string
 
 	var mutex sync.Mutex
 	eg, _ := errgroup.WithContext(context.Background())
-	fillResult := func(snap *ec2.Snapshot, volume *config.EBSVolume) {
+	fillResult := func(createOutput *ec2.CreateSnapshotsOutput) {
 		mutex.Lock()
 		defer mutex.Unlock()
-		snapIDMap[volume.ID] = *snap.SnapshotId
+		for j := range createOutput.Snapshots {
+			snapshot := createOutput.Snapshots[j]
+			snapIDMap[*snapshot.VolumeId] = *snapshot.SnapshotId
+		}
 	}
 
-	workerPool := utils.NewWorkerPool(e.concurrency, "create snapshot")
+	workerPool := utils.NewWorkerPool(e.concurrency, "create snapshots")
 	for i := range backupInfo.TiKVComponent.Stores {
 		store := backupInfo.TiKVComponent.Stores[i]
 		volumes := store.Volumes
-		if len(volumes) > 1 {
-			// if one store has multiple volume, we should respect the order
-			// raft log/engine first, then kv db. then wal
-			sort.SliceStable(volumes, func(i, j int) bool {
-				if strings.Contains(volumes[i].Type, "raft") {
-					return true
-				}
-				if strings.Contains(volumes[j].Type, "raft") {
-					return false
+		if len(volumes) >= 1 {
+			log.Info("fetch EC2 instance id using first volume")
+			var targetVolumeIDs []*string
+			for j := range volumes {
+				volume := volumes[j]
+				targetVolumeIDs = append(targetVolumeIDs, &volume.ID)
```

Comment:
looks like keep one of `targetVolumeIDs` or `volumeIDs` is enough

Reply:
targetVolumeIDs are for current tikv store, while volumeIDs are for all tikv stores.